effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -44,6 +44,11 @@ object Namer extends Phase[Parsed, NameResolved] {
 
   /** Shadow stack of modules currently named, for detection of cyclic imports */
   private val currentlyNaming: DynamicVariable[List[ModuleDecl]] = DynamicVariable(List())
+  /** Current parent top-level definition (if any) */
+  private val currentDefinition: DynamicVariable[Option[Def]] = DynamicVariable(None)
+  /** Counter to disambiguate hole identifiers */
+  private var holeCount: mutable.HashMap[String, Int] = mutable.HashMap[String, Int]()
```

Comment:
`DynamicVariable` instead of the `var`?

Reply:
It could also be a `val` (and be cleared instead of re-initialized).
Do we ever actually re-bind it (other than in `resolve(Module)`)?

It could also be a local `val` in a custom context object that gets passed implicitly. Something like:
```scala
case class NamerContext(holeCount: mutable.HahsMap[String, Int])
//...
def ... (using Context, NamerContext): ...
```

Reply:
Changed to `val` and `clear`